### PR TITLE
Inventory refactor

### PR DESCRIPTION
**Pull request by Dalt0nik**
- Remove Inventory model
- Refactor existing endpoints
- Add Patch methods to implement quantity adjustment
- Add user Merchant validation to all endpoints

### CODE REVIEW

**Review by HubertasVin — APPROVED** [review]

File: 2lab/POS/src/main/java/com/team1206/pos/user/user/UserService.java [hunk 1]
```diff
@@ -161,6 +161,10 @@ private UserRoles getCurrentUserRole() {
         return getCurrentUser().getRole();
     }
 
+    public boolean isSuperAdmin() {
+        return getCurrentUserRole() == UserRoles.SUPER_ADMIN;
+    }
+
```
Comment (Motiejukaas) [inline 2lab/POS/src/main/java/com/team1206/pos/user/user/UserService.java:167]: cia sakyciau geriau padaryti verifyRole(UserRoles userRole), kad kitom rolem nereiketu dar vieno metodo

Comment (Dalt0nik) [reply 2lab/POS/src/main/java/com/team1206/pos/user/user/UserService.java]: Pakeičiau į isRole

File: 2lab/POS/src/main/java/com/team1206/pos/inventory/productCategory/ProductCategoryService.java [hunk 2]
```diff
@@ -16,31 +18,48 @@
...
+    public ProductCategoryService(ProductCategoryRepository productCategoryrepository, MerchantRepository merchantRepository, UserService userService) {
         this.productCategoryRepository = productCategoryrepository;
         this.merchantRepository = merchantRepository;
+        this.userService = userService;
     }
 
 
     public ProductCategoryResponseDTO createProductCategory(CreateProductCategoryRequestDTO requestDTO) {
+
         Merchant merchant = merchantRepository.findById(requestDTO.getMerchantId()) // TODO pakeisti i getMerchantEntityById is merchantService
```
Comment (Motiejukaas) [inline 2lab/POS/src/main/java/com/team1206/pos/inventory/productCategory/ProductCategoryService.java:32]: cia dar TODO, nezinau ar pamirsai tiesiog

File: 2lab/POS/src/main/java/com/team1206/pos/user/user/UserService.java [hunk 3]
```diff
@@ -161,6 +161,10 @@ private UserRoles getCurrentUserRole() {
         return getCurrentUser().getRole();
     }
 
+    public boolean isRole(UserRoles role) {
```
Comment (Motiejukaas) [inline 2lab/POS/src/main/java/com/team1206/pos/user/user/UserService.java:164]: ai gali dar pervadint metoda i isCurrentUserRole, nes maisosi dabar su verifyUserRole :))

Comment (Dalt0nik) [reply 2lab/POS/src/main/java/com/team1206/pos/user/user/UserService.java:164]: 💀💀💀

**Pull request merged**